### PR TITLE
Add recipe for memprof

### PR DESCRIPTION
**Pull request by kynan**

### CODE REVIEW

**Comment by kynan** [comment]
The Travis build failure seems due to some error on their end.

File: recipes/memprof/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - cython
+    - matplotlib
```
Comment (jakirkham) [inline recipes/memprof/meta.yaml]: Is `matplotlib` needed at build time? Or can it be dropped?

Comment (kynan) [inline recipes/memprof/meta.yaml]: It's not, I'll drop it.

**Comment by kynan** [comment]
@jakirkham I've dropped the `matplotlib` build time requirement. Is this now good?

File: recipes/memprof/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8aebb0ce5ec11a42bf06c1430fb5e9e1f91aec0709056bce4af94076c853a707
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
```
Comment (jakirkham) [inline recipes/memprof/meta.yaml]: Needs `toolchain` if it has compiled code.

File: recipes/memprof/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+test:
+  imports:
+    - memprof
+  commands:
+    - python -m memprof --help
+    - mp_plot --help
+
+about:
+  home: https://jmdana.github.io/memprof/
+  license: GNU General Public License v3 (GPLv3)
```
Comment (jakirkham) [inline recipes/memprof/meta.yaml]: Please shorten to `GPL 3`.

**Comment by kynan** [comment]
@jakirkham I've made the requested changes.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**